src/frontend/src/pages/sales/SalesOrderDetail.tsx

```diff
@@ -77,6 +80,14 @@ export default function SalesOrderDetail() {
     }
   });
 
+  const orderCurrency = useMemo(() => {
```

Comment:
This seems to be needed multiple times; could this be generalised?

Reply:
The internals of the memo are slightly different. I think that generalizing this would end up with more code...